### PR TITLE
fix: only redirect if undisturbed body

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

File: lib/core/util.js [hunk 1]
```diff
@@ -224,8 +224,22 @@ function validateHandler (handler, method, upgrade) {
   }
 }
 
+function isDisturbed (body) {
```
[the remark this answers is not in the document]
Comment (ronag) [reply lib/core/util.js:229]: Done

**Comment by ronag** [comment]
@mcollina added tests and fixed bug

File: lib/handler/redirect.js [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+
+class BodyAsyncIterable {
+  constructor (body) {
+    this[kBody] = body
+    this[kBodyUsed] = false
+  }
+
+  async * [Symbol.asyncIterator] () {
+    if (this[kBodyUsed]) {
+      throw new TypeError('disturbed')
```
Comment (delvedor) [inline lib/handler/redirect.js:20]: I would throw a more meaningful error, this one looks a little obscure.

Comment (ronag) [reply lib/handler/redirect.js:20]: I'll change it to an assertion. It shouldn't happen.

File: test/redirect-request.js [hunk 3]
```diff
@@ -351,6 +353,27 @@ for (const factory of [
     }
   })
 
+  t.test('should not follow redirects when using ReadableStream request bodies', { skip: nodeMajor < 16 }, async t => {
```
Comment (mcollina) [inline test/redirect-request.js:356]: What will happen if node.js < 16?

**Review by mcollina — APPROVED** [review]
lgtm

File: test/redirect-request.js (hunk 3 above)
Comment (ronag) [reply test/redirect-request.js:356]: ReadableStream is only on node 16?

**Pull request merged**